options.go

```diff
@@ -1,17 +1,48 @@
-package hibpsync
+package hibp
 
 import (
 	"context"
 	"io"
 )
 
+// ProgressFunc represents a type of function that can be used to report progress of a sync operation.
+// The parameters are as follows:
+// - lowest: The lowest prefix that has been processed so far (due to concurrent operations, there is a window of
+// prefixes that are possibly being processed at the same time, "lowest" refers to the range with the lowest prefix).
+// - current: The current prefix that is being processed, i.e. for which the ProgressFunc gets invoked.
+// - to: The highest prefix that will be processed.
+// - processed: The number of prefixes that have been processed so far.
+// - remaining: The number of prefixes that are remaining to be processed.
+// The function should return an error if the operation should be aborted.
+type ProgressFunc func(lowest, current, to, processed, remaining int64) error
```

Comment:
Do we need to export this? I think it's just eye sugar right?

Reply:
Yes and no 😉 

As a sync can run quite some time (1-2h) having feedback on the progress actually seems quite important to me. 😊 

And without exporting this, the contained CLI command wouldn't be able to provide this.

Reply:
Ok makes sense